internal/k8s/statusaddress.go

```diff
@@ -14,82 +14,82 @@
 package k8s
 
 import (
-	"context"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"github.com/projectcontour/contour/internal/annotation"
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/api/networking/v1beta1"
-	clientset "k8s.io/client-go/kubernetes"
 )
 
-// IngressStatusUpdater observes informer OnAdd events and
+// StatusAddressUpdater observes informer OnAdd and OnUpdate events and
 // updates the ingress.status.loadBalancer field on all Ingress
 // objects that match the ingress class (if used).
-type IngressStatusUpdater struct {
-	Client       clientset.Interface
-	Logger       logrus.FieldLogger
-	Status       v1.LoadBalancerStatus
-	IngressClass string
+type StatusAddressUpdater struct {
+	Logger        logrus.FieldLogger
+	LBStatus      v1.LoadBalancerStatus
+	IngressClass  string
+	StatusUpdater StatusUpdater
 }
 
-func (s *IngressStatusUpdater) OnAdd(obj interface{}) {
+func (s *StatusAddressUpdater) OnAdd(obj interface{}) {
 
 	ing := obj.(*v1beta1.Ingress).DeepCopy()
+	ing.SetGroupVersionKind(v1beta1.SchemeGroupVersion.WithKind("ingress"))
 	if !annotation.MatchesIngressClass(ing, s.IngressClass) {
```

Comment:
Should do this check first to fail fast in the event this is an object we don't care about.

Reply:
We can't do the check first, because `MatchesIngressClass` requires typecasting from `interface{}` to a type that fulfills the `ObjectMeta` interface, in this case `*v1beta1.Ingress`. We'll have to keep this pattern even after I add HTTPProxy in here (which is most likely the only other type that will be added in here).